### PR TITLE
add http2 example

### PR DESCRIPTION
**Pull request by thinkerou**

### CODE REVIEW

**Comment by appleboy** [comment]
Can you also add instructions about how to generate the `server.pem` and `server.cert` file? Maybe create the `README.md` in `http2` folder.

**Comment by thinkerou** [comment]
@appleboy OK.

File: examples/http2/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## How to generate RSA private key and digital certificate
+
+1. Install Openssl
+
+Please visit https://github.com/openssl/openssl to get pkg and install.
+
+2. Generate RSA private key
+
+```sh
+$mkdir testdata
```
Comment (appleboy) [inline examples/http2/README.md]:
```
$ mkdir testdata
```

missing space between `$` and `mkdir`

File: examples/http2/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+$openssl genrsa -out ./testdata/server.key 2048
+```
+
+3. Generate digital certificate
+
+```sh
+openssl req -new -x509 -key ./testdata/server.key -out ./testdata/server.pem -days 365
```
Comment (appleboy) [inline examples/http2/README.md]: missing $

**Comment by thinkerou** [comment]
@appleboy Thanks, fixed.

**Comment by javierprovecho** [comment]
maybe a warning remainder (`log.Warning` etc) at the example code against use of the embed certificates...

**Comment by thinkerou** [comment]
Hi, @javierprovecho you mean that I should use `log.Fatal(r.RunTLS(":8080", "./testdata/server.pem", "./testdata/server.key"))`, right?

**Comment by javierprovecho** [comment]
@thinkerou no, I mean to add something like this before calling `r.Run()`

```go
log
  .New(os.Stderr, "", 0)
  .Println("WARNING: DON'T USE THE EMBED CERTS FROM THIS EXAMPLE IN PRODUCTION ENVIRONMENT, GENERATE YOUR OWN!")`
```

I want to be sure people new to https and gin know what they are doing 😅

**Comment by thinkerou** [comment]
@jasonrhansen Thanks, fixed #1009.

**Pull request merged**